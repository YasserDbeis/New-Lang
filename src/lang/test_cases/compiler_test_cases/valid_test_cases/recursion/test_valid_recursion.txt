func main()
{
    int x = running_sum(10);
    print(x);
}

func running_sum(int x) -> int
{
    if (x is 0)
    {
        return 0;
    }
    int running_sum = x + running_sum(x - 1);

    print("x = " + x + ", running sum = " + running_sum);
    return running_sum;
}