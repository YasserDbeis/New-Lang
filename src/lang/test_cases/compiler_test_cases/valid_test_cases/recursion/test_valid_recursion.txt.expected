x = 1, running sum = 1
x = 2, running sum = 3
x = 3, running sum = 6
x = 4, running sum = 10
x = 5, running sum = 15
x = 6, running sum = 21
x = 7, running sum = 28
x = 8, running sum = 36
x = 9, running sum = 45
x = 10, running sum = 55
55